c/tskit/trees.c

```diff
@@ -4551,47 +4551,95 @@ tsk_tree_position_in_interval(const tsk_tree_t *self, double x)
     return self->interval.left <= x && x < self->interval.right;
 }
 
+static int
+tsk_tree_seek_from_null(tsk_tree_t *self, double x, tsk_flags_t TSK_UNUSED(options))
+{
+    int ret = 0;
+    tsk_size_t edge, num_edges = self->tree_sequence->tables->edges.num_rows,
+                     num_trees = self->tree_sequence->num_trees;
+    tsk_id_t p, c, e, tree_index;
+    const double L = tsk_treeseq_get_sequence_length(self->tree_sequence);
+    const tsk_id_t *restrict edge_parent = self->tree_sequence->tables->edges.parent;
+    const tsk_id_t *restrict edge_child = self->tree_sequence->tables->edges.child;
+    const double *restrict edge_left = self->tree_sequence->tables->edges.left;
+    const double *restrict edge_right = self->tree_sequence->tables->edges.right;
+    const double *restrict breakpoints = self->tree_sequence->breakpoints;
+    const tsk_id_t *restrict insertion
+        = self->tree_sequence->tables->indexes.edge_insertion_order;
+    const tsk_id_t *restrict removal
+        = self->tree_sequence->tables->indexes.edge_removal_order;
+
+    /* NOTE: this differ's from JK's proposal by not using
+     * edge diffs explicitly. However:
+     * 1. It benchmarked faster (in rust).
+     * 2. It is a lot less code and doesn't have
+     * to do "edge diffs, but backwards".
+     */
+    self->num_edges = 0;
+
+    if (x <= L / 2.0) {
+        for (edge = 0; edge < num_edges; ++edge) {
+            e = insertion[edge];
+            if (edge_left[e] > x) {
+                break;
+            }
+            if (x >= edge_left[e] && x < edge_right[e]) {
+                p = edge_parent[e];
+                c = edge_child[e];
+                tsk_tree_insert_edge(self, p, c, e);
+            }
+        }
+    } else {
+        for (edge = 0; edge < num_edges; ++edge) {
```

Comment:
I'm not sure there's actually much point in this - we've documented that users shouldn't depend on the specific order of children within a node (which is what changes when you apply/remove edges in different orders) and anyway, it *still* won't be the same even if we seek in the same direction as the standard code. 

So, I think completely *identical* quintuply linked trees is a non-goal here, because it's essentially impossible.

Reply:
Sorry--not sure what is being discussed here?  The code that is highlighted appears unrelated to the comments?  If you are referring to the two seek directions, performance is the goal -- see my previous figures.  We don't want that positive slope in the index v. time plot, which results from always iteratiing from one direction.

Reply:
Ah, right, forgot about the ``break`` above. I get it now, ignore!